code/modules/surgery/organs/lungs.dm

```diff
@@ -356,38 +47,30 @@
 	. = TRUE
 	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
 		return
-	switch(species_state)
-		if("human") // from human to vox
-			safe_oxygen_min = 0
-			safe_oxygen_max = safe_toxins_max
-			safe_nitro_min = 16
-			oxy_damage_type = TOX
-			to_chat(user, "<span class='notice'>You configure [src] to replace vox lungs.</span>")
-			species_state = "vox"
-		if("vox") // from vox to plasmamen
-			safe_oxygen_max = initial(safe_oxygen_max)
-			safe_toxins_min = 16
-			safe_toxins_max = 0
-			safe_nitro_min = initial(safe_nitro_min)
-			oxy_damage_type = OXY
-			to_chat(user, "<span class='notice'>You configure [src] to replace plasmamen lungs.</span>")
-			species_state = "plasmamen"
-		if("plasmamen") // from plasmamen to human
-			safe_oxygen_min = initial(safe_oxygen_min)
-			safe_toxins_min = initial(safe_toxins_min)
-			safe_toxins_max = initial(safe_toxins_max)
-			to_chat(user, "<span class='notice'>You configure [src] back to default settings.</span>")
-			species_state = "human"
+
+	var/possible = list("default" = /datum/organ/lungs, "vox" = /datum/organ/lungs/vox, "plasmamen" = /datum/organ/lungs/plasmamen)
+	var/chosen = input(user, "Select lung type", "What kind of lung settings?") as null|anything in possible
+	if(isnull(chosen) || chosen == species_state)
+		return
```

Comment:
you need to proximity check after chosen and also make sure they still have the multitool !!!

Reply:
Yeah just need to do this, other reviews are done but unpushed

Reply:
Proximity check done, how would you suggest the multitool check? Just another `I.use_tool(src, user, 0, volume = I.tool_volume)`?

Reply:
yeah sure

Reply:
Done